hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/EndpointBase.java

```diff
@@ -290,8 +290,12 @@ protected Map<String, String> getCustomMetadataFromHeaders(
 
         if (sizeInBytes >
                 OzoneConsts.S3_REQUEST_HEADER_METADATA_SIZE_LIMIT_KB * KB) {
-          throw new IllegalArgumentException("Illegal user defined metadata." +
-              " Combined size cannot exceed 2KB.");
+          OS3Exception os3Exception = newError(S3ErrorTable.METADATA_TOO_LARGE,
+              key);
+          os3Exception.setErrorMessage("Illegal user defined metadata. " +
```

Comment:
`setErrorMessage` here is nonsense, which completely overlays the errorMessage in S3ErrorTable.METADATA_TOO_LARGE. 

The Description from AWS is not necessary to be strictly followed, the Error Code should be.